recipes/autoconf/all/conanfile.py

```diff
@@ -13,110 +18,89 @@ class AutoconfConan(ConanFile):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def build(self):
+        apply_conandata_patches(self)
+        autotool = Autotools(self)
+        autotool.configure()
+        autotool.make()
+        subsystem = deduce_subsystem(self, scope="build")
+        install_path = subsystem_path(subsystem, self.package_folder)
+        autotool.install(args=[f"DESTDIR={install_path}"])
+
+    def package(self):
+        packager = AutoPackager(self)
+        packager.run()
+        rmdir(self, Path(self.package_folder, "share", "info"))
+        rmdir(self, Path(self.package_folder, "share", "man"))
 
     @property
     def _datarootdir(self):
-        return os.path.join(self.package_folder, "bin", "share")
+        return Path(self.package_folder, "share")
 
     @property
     def _autoconf_datarootdir(self):
-        return os.path.join(self._datarootdir, "autoconf")
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        datarootdir = self._datarootdir
-        prefix = self.package_folder
-        if self.settings.os == "Windows":
-            datarootdir = tools.unix_path(datarootdir)
-            prefix = tools.unix_path(prefix)
-        conf_args = [
-            "--datarootdir={}".format(datarootdir),
-            "--prefix={}".format(prefix),
-        ]
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
-
-    def _patch_files(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
-    @contextlib.contextmanager
-    def _build_context(self):
-        with tools.environment_append(tools.RunEnvironment(self).vars):
-            yield
-
-    def build(self):
-        self._patch_files()
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make()
-
-    def package(self):
-        self.copy("COPYING*", src=self._source_subfolder, dst="licenses")
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.install()
-        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "info"))
-        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "man"))
+        return Path(self._datarootdir, "autoconf")
 
     def package_info(self):
-        self.cpp_info.libdirs = []
-        self.cpp_info.includedirs = []
+        subsystem = deduce_subsystem(self, scope="build")
 
-        bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH env var with : {}".format(bin_path))
-        self.env_info.PATH.append(bin_path)
```

Comment:
Can not be removed. env_info.PATH is still required due Conan 1.x generators.

Reply:
I thought the `layout` method took care of this, since it defines the `bindirs` which is automatically added to the `PATH`.

Will revert this change, can't hurt if this is added twice I think